library/src/main/scala/scala/scalajs/js/BigInt.scala

```diff
@@ -0,0 +1,116 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package scala.scalajs.js
+
+import scala.scalajs.js
+import scala.scalajs.js.annotation.JSGlobal
+
+/** <span class="badge badge-ecma2019" style="float: right;">ECMAScript 2019</span>
+ *
+ *  A built-in object that provides a way to represent whole numbers larger than
+ *  2 ^ 53 - 1, which is the largest number JavaScript can reliably represent
+ *  with the Number primitive.
+ *
+ *  BigInt can be used for arbitrarily large integers.
+ *
+ *  @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt
+ */
+@js.native
+@JSGlobal
+final class BigInt private[this] () extends js.Object {
+
+  def +(other: BigInt): BigInt = js.native
+  def *(other: BigInt): BigInt = js.native
+  def -(other: BigInt): BigInt = js.native
+  def **(other: BigInt): BigInt = js.native
+  def %(other: BigInt): BigInt = js.native
+
+  def &(other: BigInt): BigInt = js.native
+  def |(other: BigInt): BigInt = js.native
+  def ^(other: BigInt): BigInt = js.native
+  def <<(other: BigInt): BigInt = js.native
+  def >>(other: BigInt): BigInt = js.native
+  // no >>> since BigInt is always signed
+
+  // scalastyle:off disallow.space.before.token
+  def unary_- : BigInt = js.native
+  def unary_~ : BigInt = js.native
+  // no unary_+ not to break asm.js
```

Comment:
The rationale "not to break asm.js" is meaningful in the spec of `BigInt`s, but here it is misleading as it suggests that *Scala.js* doesn't want to break asm.js and therefore does not expose `unary_+`.
```suggestion
  // unary_+ is not supported by BigInts
```